lib/tpm2_hmac_auth.c

```diff
@@ -0,0 +1,92 @@
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+#ifndef HMAC_AUTH_H
+#define HMAC_AUTH_H
+
+#include <string.h>
+
+#include <tss2/tss2_sys.h>
+
+#include "log.h"
+#include "tpm2_error.h"
+#include "tpm2_nv_util.h"
+
+bool tpm2_read_transient_persistent_obj_name(TSS2_SYS_CONTEXT *sapi_context,
+    TPM2_HANDLE entity_handle, TPM2B_NAME *entity_name) {
+    TPM2B_PUBLIC object_public;
+    TSS2L_SYS_AUTH_RESPONSE sessions_data_out;
+    TPM2B_NAME qualified_name = TPM2B_TYPE_INIT(TPM2B_NAME, name);
+    TSS2_RC rval = TSS2_RETRY_EXP( Tss2_Sys_ReadPublic(sapi_context, entity_handle, 
+                                        NULL, &object_public, entity_name, &qualified_name, 
+                                        &sessions_data_out));
+    if (rval != TPM2_RC_SUCCESS) {
+        LOG_PERR(Tss2_Sys_ReadPublic, rval);
+        return false;
+    }
+    return true;
+}
+
+bool tpm2_get_entity_name(TSS2_SYS_CONTEXT *sapi_context, TPM2_HANDLE entity_handle, 
+    TPM2B_NAME *entity_name) {
+    TPM2B_NV_PUBLIC nv_public = TPM2B_EMPTY_INIT;
+    bool res;
+    uint8_t temp[4];
+    //Get Entity names from handles
+    switch(entity_handle & 0xFF000000) {
+        case 0x01000000: res = tpm2_util_nv_read_public(sapi_context, 
+                                                        entity_handle, 
+                                                        &nv_public,
+                                                        entity_name);
+                    if (!res) {
+                        LOG_ERR("Failed reading NV public when calculating name");
+                        return false;
+                    }
+                    break;
+        case 0x80000000:
+        case 0x81000000: res = tpm2_read_transient_persistent_obj_name(sapi_context,
+                                    entity_handle, entity_name);
+                         if (!res) {
+                            LOG_ERR("Failed reading Object public when calculating name");
+                            return false;
+                        }
+                        break;
+        case 0x00000000:
+        case 0x02000000:
+        case 0x03000000:
+        case 0x40000000: entity_name->size = sizeof(TPM2_HANDLE);
+                         memcpy(temp, (uint8_t *)&entity_handle, sizeof(TPM2_HANDLE));
+                         for(unsigned i=0; i<sizeof(TPM2_HANDLE); i++) {
```

Comment:
Please use a space between the operator and operands.

Reply:
fixed